docarray/index/backends/redis.py

```diff
@@ -0,0 +1,609 @@
+        }
+
+        for py_type, redis_type in type_map.items():
+            if issubclass(python_type, py_type):
+                return redis_type
+        raise ValueError(f'Unsupported column type for {type(self)}: {python_type}')
+
+    @staticmethod
+    def _generate_items(
+        column_to_data: Dict[str, Generator[Any, None, None]],
+        batch_size: int,
+    ) -> Iterator[List[Dict[str, Any]]]:
+        """
+        Given a dictionary of generators, yield a list of dictionaries where each
+        item consists of a key and a single item from the corresponding generator.
+
+        :param column_to_data: A dictionary where each key is a column and each value
+            is a generator.
+        :param batch_size: Size of batch to generate each time.
+
+        :yield: A list of dictionaries where each item consists of a column name and
+            an item from the corresponding generator. Yields until all generators
+            are exhausted.
+        """
+        keys = list(column_to_data.keys())
+        iterators = [iter(column_to_data[key]) for key in keys]
+        batch: List[Dict[str, Any]] = []
+
+        while True:
+            item_dict = {}
+            for key, it in zip(keys, iterators):
+                item = next(it, None)
+
+                if key == 'id' and not item:
+                    if batch:
+                        yield batch
+                    return
+
+                if isinstance(item, AbstractTensor):
+                    item_dict[key] = item._docarray_to_ndarray().tolist()
+                elif isinstance(item, ndarray):
+                    item_dict[key] = item.astype(np.float32).tolist()
+                elif item is not None:
+                    item_dict[key] = item
+
+            batch.append(item_dict)
+            if len(batch) == batch_size:
+                yield batch
+                batch = []
+
+    def _index(
+        self, column_to_data: Dict[str, Generator[Any, None, None]]
+    ) -> List[str]:
+        """
+        Indexes the given data into Redis.
+
+        :param column_to_data: A dictionary where each key is a column and each value is a generator.
+        :return: A list of document ids that have been indexed.
+        """
+        self._index_subindex(column_to_data)
+        ids: List[str] = []
+        for items in self._generate_items(
+            column_to_data, self._runtime_config.batch_size
+        ):
+            doc_id_item_pairs = [
+                (self._prefix + item['id'], '$', item) for item in items
+            ]
+            ids.extend(doc_id for doc_id, _, _ in doc_id_item_pairs)
+            self._client.json().mset(doc_id_item_pairs)  # type: ignore[attr-defined]
+
+        return ids
+
+    def num_docs(self) -> int:
+        """
+        Fetch the number of documents in the index.
+
+        :return: Number of documents in the index.
+        """
+        num_docs = self._client.ft(self._index_name).info()['num_docs']
```

Comment:
It is, `num_docs` is included in the index info

but apparently there's a better way to get number of docs
`        return self._client.dbsize()
`

Reply:
But I think we should stick with the old way because dbsize returns number of keys the client stores, combining all indices so it's inaccurate if you have more than one index